kedro/contrib/io/dask/dask_parquet_s3.py

```diff
@@ -0,0 +1,116 @@
+# trademarks of QuantumBlack. The License does not grant you any right or
+# license to the QuantumBlack Trademarks. You may not use the QuantumBlack
+# Trademarks or any confusingly similar mark as a trademark for your product,
+#     or use the QuantumBlack Trademarks in any other manner that might cause
+# confusion in the marketplace, including but not limited to in advertising,
+# on websites, or on software.
+#
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""``ParquetS3DaskDataSet`` is a data set used to load and save
+data to parquet files using Dask on S3
+"""
+from typing import Any, Dict, Optional
+
+import dask.dataframe as dd
+
+from kedro.contrib.io import DefaultArgumentsMixIn
+from kedro.io.core import AbstractDataSet, DataSetError
+
+
+class ParquetS3DaskDataSet(DefaultArgumentsMixIn, AbstractDataSet):
+    """``ParquetS3DaskDataSet`` loads and saves data to file(s) in S3. It uses s3fs
+        to read and write from S3 and dask to handle the parquet file.
+
+        Example:
+        ::
+
+            >>> from kedro.contrib.io.parquet.dask_parquet_s3 import ParquetS3DaskDataSet
+            >>> import pandas as pd
+            >>> import dask.dataframe as dd
+            >>>
+            >>> data = pd.DataFrame({'col1': [1, 2], 'col2': [4, 5],
+            >>>                      'col3': [5, 6]})
+            >>> ddf = dd.from_pandas(data, npartitions=2)
+            >>>
+            >>> data_set = ParquetS3DaskDataSet(
+            >>>                         filepath="temp_folder",
+            >>>                         credentials={
+            >>>                             'aws_access_key_id': 'YOUR_KEY',
+            >>>                             'aws_secret_access_key': 'YOUR SECRET'},
+            >>>                         save_args={"compression": "GZIP"})
+            >>> data_set.save(data)
+            >>> reloaded = data_set.load()
+            >>>
+            >>> assert ddf.compute().equals(reloaded.compute())
+    """
+
+    # pylint: disable=too-many-arguments
+    def __init__(
+        self,
+        filepath: str,
+        credentials: Optional[Dict[str, Any]] = None,
+        load_args: Optional[Dict[str, Any]] = None,
+        save_args: Optional[Dict[str, Any]] = None,
+    ) -> None:
+        """Creates a new instance of ``ParquetS3DaskDataSet`` pointing to a concrete
+        parquet file on S3.
+
+        Args:
+            filepath: Path to a parquet file
+                parquet collection or the directory of a multipart parquet.
+            bucket_name: S3 bucket name.
+            credentials: Credentials to access the S3 bucket, such as
+                ``aws_access_key_id``, ``aws_secret_access_key``.
+            load_args: Additional loading options `pyarrow`:
+                https://arrow.apache.org/docs/python/generated/pyarrow.parquet.read_table.html
```

Comment:
Since you pass `load_args` and `save_args` to Dask `read_parquet` and `to_parquet` respectively (rather than `pyarrow`), I would suggest to put the links to the respective Dask APIs